### ISSUE TITLE
Some automated test for the library

### ISSUE THREAD

**Issue by jedbrown** [barbagroup/AmgXWrapper#19]
AmgXWrapper has two distribution modes: as source files to be compiled by the client and as a shared library. Bundling the source files is perhaps not too onerous for an application, but wholly inappropriate for another library -- libraries cannot assume that some other part of the application won't also call AmgXWrapper. As far as I can tell, there are no examples or automated tests for the library. It's also rather complicated to determine if things are working correctly upon building the library -- you need to go out into a different directory and configure a different build (probably with the same command-line options) to build the example, then run it and parse the output (which has some terminal noise; e.g., about JSON vs AmgX input file formats). There should at least be a "make test" target that for the library that verifies that the build is working correctly.

**Comment by piyueh** [comment]
Agree. We will create some automatic tests and make both examples use the library.
